resolver.js

```diff
@@ -33,45 +34,68 @@ var domainForResolver = function domainForResolver(req) {
 }
 
 var defaultResolver = function resolver(host, url, req) {
+    var wikiDomain = domainForResolver(req)
+    if (wikiDomain === null) {
+        return Promise.resolve(null)
+    }
+
+    return fetchWikiForDomain(wikiDomain)
+        .then(function (response) {
+            if (!response || !response.data) {
+                return null
+            }
+
+            let backend = response.data.wiki_queryservice_namespace.backend
+            let namespace = response.data.wiki_queryservice_namespace.namespace
+            var parsed = parse('http://' + host + url, true)
+            parsed.set('pathname', '/bigdata/namespace/' + namespace)
+            parsed.set('hostname', backend)
+            // unset the port as it is set in the backend host...
+            parsed.set('port', '')
+
+            return parsed.toString()
+        })
+        .catch(function (err) {
+            console.error(err)
+            throw err
+        })
+
+}
 
-    // Try and log on failures (so this process keeps running...)
-    try {
+var responseCache = new LRUCache({
+    ttl: parseInt(process.env.RESPONSE_CACHE_TTL || 60 * 60, 10) * 1000
```

Comment:
This needs a chart update. Any opinions on how long to cache such an item? I'd think 60 minutes is a good value to have.

Reply:
PR here https://github.com/wbstack/charts/pull/118